evalml/model_understanding/feature_explanations.py

```diff
@@ -0,0 +1,208 @@
+"""Human Readable Pipeline Explanations."""
+
+from evalml.model_family import ModelFamily
+from evalml.model_understanding import calculate_permutation_importance
+from evalml.utils import get_logger, infer_feature_types
+
+
+def readable_explanation(
+    pipeline,
+    X=None,
+    y=None,
+    importance_method="permutation",
+    max_features=5,
+    min_importance_threshold=0.05,
+    objective="auto",
+):
+    """Outputs a human-readable explanation of trained pipeline behavior.
+
+    Args:
+        pipeline (PipelineBase): The pipeline to explain.
+        X (pd.DataFrame): If importance_method is permutation, the holdout X data to compute importance with. Ignored otherwise.
+        y (pd.Series): The holdout y data, used to obtain the name of the target class. If importance_method is permutation, used to compute importance with.
+        importance_method (str): The method of determining feature importance. One of ["permutation", "feature"]. Defaults to "permutation".
+        max_features (int): The maximum number of features to include in an explanation. Defaults to 5.
+        min_importance_threshold (float): The minimum percent of total importance a single feature can have to be considered important. Defaults to 0.05.
+        objective (str, ObjectiveBase): If importance_method is permutation, the objective to compute importance with. Ignored otherwise, defaults to "auto".
+
+    Raises:
+        ValueError: if any arguments passed in are invalid or the pipeline is not fitted.
+    """
+    logger = get_logger(f"{__name__}.explain")
+
+    if not pipeline._is_fitted:
+        raise ValueError(
+            "Pipelines must be fitted in order to run feature explanations."
+        )
+
+    if min_importance_threshold >= 1 or min_importance_threshold < 0:
+        raise ValueError(
+            f"The minimum importance threshold must be a percentage value in the range [0, 1), not {min_importance_threshold}."
+        )
+
+    if importance_method == "permutation":
+
+        if objective == "auto":
```

Comment:
Ah, I didn't realize that existed! I think for now I'm going to leave this though, as it allows for keeping the objective names as lowercase rather than title case, and importing `get_default_primary_search_objective` causes circular imports.